BUG: github action occationally fails

The error message reads:
> PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'tests\\data\\locations_plot.png'

or 

> assert os.path.exists(tmp_file)      
AssertionError: assert False

Guess it is due to the same `tmp_file` name for all plotting functions that lead to IO conflict during testing. 

We can either test the save file only once in the plotting or change their `tmp_file` name.

--- a/tests/visualization/test_locations.py
+++ b/tests/visualization/test_locations.py
@@ -29,7 +29,7 @@ class TestPlot_locations:
     def test_locations_plot(self, test_data):
         """Use trackintel visualization function to plot locations and check if the file exists."""
         pfs, stps, locs = test_data
-        tmp_file = os.path.join("tests", "data", "locations_plot.png")
+        tmp_file = os.path.join("tests", "data", "locations_plot1.png")
         locs.as_locations.plot(
             out_filename=tmp_file, radius=200, positionfixes=pfs, staypoints=stps, staypoints_radius=100, plot_osm=False
         )
@@ -39,7 +39,7 @@ class TestPlot_locations:
     def test_axis(self, test_data):
         """Test the use of regular_figure() to create axis."""
         pfs, _, locs = test_data
-        tmp_file = os.path.join("tests", "data", "locations_plot.png")
+        tmp_file = os.path.join("tests", "data", "locations_plot2.png")
         _, ax = regular_figure()
 
         locs.as_locations.plot(
@@ -54,8 +54,8 @@ class TestPlot_locations:
 
     def test_parameter(self, test_data):
         """Test other parameter configurations."""
-        pfs, _, locs = test_data
-        tmp_file = os.path.join("tests", "data", "locations_plot.png")
+        _, _, locs = test_data
+        tmp_file = os.path.join("tests", "data", "locations_plot3.png")
 
         # plot only location
         locs.as_locations.plot(out_filename=tmp_file, plot_osm=True)

--- a/tests/visualization/test_staypoints.py
+++ b/tests/visualization/test_staypoints.py
@@ -23,7 +23,7 @@ class TestPlot_staypoints:
     def test_staypoints_plot(self, test_data):
         """Use trackintel visualization function to plot staypoints and check if the file exists."""
         pfs, stps = test_data
-        tmp_file = os.path.join("tests", "data", "staypoints_plot.png")
+        tmp_file = os.path.join("tests", "data", "staypoints_plot1.png")
 
         stps.as_staypoints.plot(out_filename=tmp_file, radius=100, positionfixes=pfs, plot_osm=False)
         assert os.path.exists(tmp_file)
@@ -32,7 +32,7 @@ class TestPlot_staypoints:
     def test_parameter(self, test_data):
         """Test other parameter configurations."""
         pfs, stps = test_data
-        tmp_file = os.path.join("tests", "data", "staypoints_plot.png")
+        tmp_file = os.path.join("tests", "data", "staypoints_plot2.png")
 
         # no radius
         stps.as_staypoints.plot(out_filename=tmp_file, positionfixes=pfs, plot_osm=False)
@@ -40,6 +40,6 @@ class TestPlot_staypoints:
         os.remove(tmp_file)
 
         # with osm
-        stps.as_staypoints.plot(out_filename="staypoints_plot", plot_osm=True)
-        assert os.path.exists("staypoints_plot.png")
-        os.remove("staypoints_plot.png")
+        stps.as_staypoints.plot(out_filename=tmp_file, plot_osm=True)
+        assert os.path.exists(tmp_file)
+        os.remove(tmp_file)

--- a/tests/visualization/test_triplegs.py
+++ b/tests/visualization/test_triplegs.py
@@ -27,7 +27,7 @@ class TestPlot_triplegs:
         """Use trackintel visualization function to plot triplegs and check if the file exists."""
         pfs, stps, tpls = test_data
 
-        tmp_file = os.path.join("tests", "data", "triplegs_plot.png")
+        tmp_file = os.path.join("tests", "data", "triplegs_plot1.png")
         tpls.as_triplegs.plot(out_filename=tmp_file, positionfixes=pfs, staypoints=stps, plot_osm=False)
         assert os.path.exists(tmp_file)
         os.remove(tmp_file)
@@ -35,7 +35,7 @@ class TestPlot_triplegs:
     def test_axis(self, test_data):
         """Test the use of regular_figure() to create axis."""
         _, _, tpls = test_data
-        tmp_file = os.path.join("tests", "data", "triplegs_plot.png")
+        tmp_file = os.path.join("tests", "data", "triplegs_plot2.png")
         _, ax = regular_figure()
 
         tpls.as_triplegs.plot(out_filename=tmp_file, axis=ax)
@@ -45,7 +45,7 @@ class TestPlot_triplegs:
     def test_parameter(self, test_data):
         """Test other parameter configurations."""
         _, _, tpls = test_data
-        tmp_file = os.path.join("tests", "data", "triplegs_plot.png")
+        tmp_file = os.path.join("tests", "data", "triplegs_plot3.png")
 
         # test plot_osm
         tpls.as_triplegs.plot(out_filename=tmp_file, plot_osm=True)

